src/components/checkbox/GroupableCheckboxReducersUtils.ts

```diff
@@ -0,0 +1,171 @@
+import * as _ from 'underscore';
+import {IReduxAction} from '../../utils/ReduxUtils';
+import {ICheckboxState} from './CheckboxReducers';
+import {IGroupableCheckboxActionPayload} from './GroupableCheckboxActions';
+import {groupableCheckboxInitialState, IGroupableCheckboxesState} from './GroupableCheckboxReducers';
+
+export const calculateTotalCheckboxesChecked = (checkboxes: ICheckboxState[]): number => _.where(checkboxes, {checked: true}).length;
+
+export const addParentCheckbox = (state: IGroupableCheckboxesState, action: IReduxAction<IGroupableCheckboxActionPayload>): IGroupableCheckboxesState => {
+    const newCheckbox = {
+        id: action.payload.id,
+        checked: !!action.payload.checked,
+        disabled: !!action.payload.disabled,
+    };
+
+    if (state === groupableCheckboxInitialState) {
+        return {
+            ...state,
+            parentId: action.payload.id,

```

Comment:
thinking outloud, do you absolutely need `parentId` if you have it thourgh `parent.id` ?

Reply:
this variable have 2 functionalities, give an id for the groupableCheckbox and to handle the problem. When its time to add each checkbox in the state.,I'm not sure the parent will be added first. I created this variable to simplify the process for the add.